llvm/lib/Target/AArch64/AArch64TargetTransformInfo.cpp

```diff
@@ -568,6 +568,64 @@ AArch64TTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     }
     return Cost;
   }
+  case Intrinsic::vector_extract: {
+    // If both the vector argument and the return type are legal types and the
+    // index is 0, then this should be a no-op or simple operation; return a
+    // relatively low cost.
+
+    // If arguments aren't actually supplied, then we cannot determine the
+    // value of the index. We also want to skip this on predicate types.
+    if (ICA.getArgs().size() != 2 ||
+        ICA.getReturnType()->getScalarType()->isIntegerTy(1))
+      break;
+    LLVMContext &C = RetTy->getContext();
+    EVT MRTy = getTLI()->getValueType(DL, RetTy);
+    EVT MPTy = getTLI()->getValueType(DL, ICA.getArgTypes()[0]);
+    // Skip this if either the return type or the vector argument are unpacked
+    // SVE types; they may get lowered to stack stores and loads.
+    if ((MRTy.isScalableVector() &&
+         MRTy.getSizeInBits().getKnownMinValue() != AArch64::SVEBitsPerBlock) ||
+        (MPTy.isScalableVector() &&
+         MPTy.getSizeInBits().getKnownMinValue() != AArch64::SVEBitsPerBlock))
```

Comment:
```suggestion
         MPTy.getSizeInBits().getKnownMinValue() < AArch64::SVEBitsPerBlock))
```
Do you want to use a less-than comparison instead? Extracting a `<vscale x 4 x float>` from a `<vscale x 8 x float>` from index 0 is free. With this code it isn't free, but it's probably a good and common case to add?

Can you also create a `isUnpackedVectorVT` function to make this code a bit simpler?